packages/transformers/babel/src/BabelTransformer.js

```diff
@@ -3,29 +3,27 @@ import {Transformer} from '@parcel/plugin';
 import SourceMap from '@parcel/source-map';
 import generate from '@babel/generator';
 import semver from 'semver';
-import babel6 from './babel6';
 import babel7 from './babel7';
-import getBabelConfig from './config';
 import {relativeUrl} from '@parcel/utils';
+import {load, rehydrate} from './config';
 
 export default new Transformer({
-  async getConfig({asset}) {
-    return getBabelConfig(asset);
+  async loadConfig(config) {
```

Comment:
Can we make these method signatures match the others with an options object? That way we can add other params in the future more easily.

Also, do you think these methods should have access to the Asset being transformed, and the Parcel options object?

Reply:
Options I'll definitely start passing now. The asset seems like it could be useful especially if the content of the asset might influence the config, so we don't have to read the file more than once. It's a little tricky right now because the loading of the config is taking place in the main process which doesn't have access to the asset and I'm not sure it would be a good idea to pass it across IPC. I've been thinking of ways that we could maybe get rid of the reverse handles, so if we figure that out that would enable us to provide the asset. One downside of not using reverse handles anymore would be when we do need to reload config, internal tool caches would be spread out across all of the processes. Probably still worth it, though.

For now plugins at least have access to the assets path. Currently it's named `searchPath`, but a better name for that may be `assetPath`.